src/averageAges.js

```diff
@@ -1,61 +1,57 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const whatWeFiler = people.filter(person => (
+    arguments.length === 1
+      ? person.sex === 'm'
+      : person.sex === 'm' && Math.ceil(person.died / 100) === century
+  ));
+  const ageDiffArray = [];
+
+  whatWeFiler.forEach(person => {
+    ageDiffArray.push(person.died - person.born);
+  });
```

Comment:
you repeated this logic 2 times for different arrays
let's create separate function that will accept people array and will return the average people age